### PR TITLE
Add GetTransactionInfo in VTadmin API and page

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR adds a VTAdmin API for getting the transaction information added in https://github.com/vitessio/vitess/pull/17131. This PR should be merged after it is https://github.com/vitessio/vitess/pull/17131.

This PR also adds a page to display the following page - 
Screenshot 2024-11-08 at 3 20 33 PM

## Related Issue(s)

- #16245 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/cluster/vtadmin_process.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2020 The Vitess Authors.
```
Comment (systay) [inline go/test/endtoend/cluster/vtadmin_process.go:2]: date

File: go/test/endtoend/vtadmin/main_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2022 The Vitess Authors.
```
Comment (systay) [inline go/test/endtoend/vtadmin/main_test.go:2]: date

**Review by systay — APPROVED** [review]

File: go/vt/vtadmin/api.go [hunk 3]
```diff
@@ -1521,6 +1522,25 @@ func (api *API) GetTopologyPath(ctx context.Context, req *vtadminpb.GetTopologyP
...
+	defer span.Finish()
+
+	c, err := api.getClusterForRequest(req.ClusterId)
+	if err != nil {
+		return nil, err
+	}
+
+	cluster.AnnotateSpan(c, span)
+
+	if !api.authz.IsAuthorized(ctx, c.ID, rbac.KeyspaceResource, rbac.GetAction) {
```
[the remark this answers is not in the document]
Comment (notfelineit) [reply go/vt/vtadmin/api.go:1537]:
```suggestion
	if !api.authz.IsAuthorized(ctx, c.ID, rbac.ClusterResource, rbac.GetAction) {
```
Cluster resource might be more accurate to your use case!

**Review by notfelineit — APPROVED** [review]
Left one comment!

File: go/test/endtoend/cluster/vtadmin_process.go (hunk 1 above)
Comment (GuptaManan100) [reply go/test/endtoend/cluster/vtadmin_process.go:2]: Fixed

File: go/test/endtoend/vtadmin/main_test.go (hunk 2 above)
Comment (GuptaManan100) [reply go/test/endtoend/vtadmin/main_test.go:2]: Fixed

**Comment by GuptaManan100** [comment]
Thank you for the reviews! I have fixed up all the comments and also the parsing of time. We were passing it in in nanoseconds, but the dayjs was expecting it to be in seconds. I have fixed this now.

**Pull request merged**